packages/core/src/test/amazonqFeatureDev/controllers/chat/controller.test.ts

```diff
@@ -471,12 +471,31 @@ describe('Controller', () => {
                 ['EmptyPatchException', MetricDataResult.LlmFailure],
                 [PromptRefusalException.name, MetricDataResult.Error],
                 [NoChangeRequiredException.name, MetricDataResult.Error],
+                [MonthlyConversationLimitError.name, MetricDataResult.Error],
+                [CodeIterationLimitError.name, MetricDataResult.Error],
             ])
 
             function getMetricResult(error: ToolkitError): MetricDataResult {
                 if (error instanceof FeatureDevServiceError && error.code) {
                     return errorResultMapping.get(error.code) ?? MetricDataResult.Error
                 }
+                if (
+                    error.code === 'StartCodeGenerationFailed' &&
+                    (error.message.includes('Improperly formed request') ||
+                        error.message.includes('Resource not found'))
+                ) {
+                    return MetricDataResult.Error
+                }
+                if (error.message.includes('StartTaskAssistCodeGeneration reached for this month.')) {
```

Comment:
This error text is also filtered in featureDev.ts inside of startCodeGeneration, might make sense to pull that out as well